packages/rspack-dev-server/src/server.ts

```diff
@@ -13,23 +13,31 @@ import type { Server } from "node:http";
 import type { Socket } from "node:net";
 import { type Compiler, MultiCompiler } from "@rspack/core";
 import type { FSWatcher } from "chokidar";
-import rdm from "webpack-dev-middleware";
-import WebpackDevServer, {
-	type OverlayMessageOptions
-} from "webpack-dev-server";
+import WebpackDevServer from "webpack-dev-server";
 // @ts-ignore 'package.json' is not under 'rootDir'
 import { version } from "../package.json";
 
+import { addResolveAlias } from "./alias";
 import type { DevServer, ResolvedDevServer } from "./config";
 import { applyDevServerPatch } from "./patch";
 
+// override
+addResolveAlias("webpack-dev-server", {
```

Comment:
Can we only override this before the call site and then remove the override after the call site, just to control the effect as small as possible and not affect other resolve

Reply:
This override only affect the require.resolve of `webpack-dev-server`. Has any other place use it too?

Reply:
Oh I see, when required `rspack-dev-server` but not use it, then starting a webpack-dev-server will be affected. I will find a better way to handle this

Reply:
Seems that only need to override before `addAdditionalEntries `, this method is sync so override before it and resume after it will make the effect as small as possible.